File: algorithms/347.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> topKFrequent(vector<int> &nums, int k) {
        unordered_map<int, int> frequencyMap;

        for (auto num : nums)
            frequencyMap[num] += 1;

        vector< vector<int> > bucket(nums.size()+1);
        for (auto it = frequencyMap.begin(); it != frequencyMap.end(); ++it)
            bucket[it->second].push_back(it->first);

        vector<int> result;
        for (int i = nums.size(); i > 0 && result.size() < k; --i) {
            result.insert(result.begin(), bucket[i].begin(), bucket[i].end());
        }
        return result;
    }
};

int main() {
    return 0;
}

File: algorithms/python/347.py
import heapq
from collections import Counter

class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        c = Counter(nums)
        c = heapq.nlargest(k, c.items(), key=lambda x: x[1])
        rls = [e[0] for e in c]
        return rls
